### PR TITLE
Add velocity templates for Streaming APIs

### PR DESCRIPTION
**Pull request by senthuran16**
## Purpose
Add velocity templates for Streaming APIs

### CODE REVIEW

File: modules/distribution/resources/api_templates/velocity_template.xml [hunk 1]
```diff
@@ -125,7 +125,16 @@
...
 
-<api xmlns="http://ws.apache.org/ns/synapse"  name="$!apiName" context="$!apiContext" version="$!apiVersion" transports="$!transport" version-type="context">
+<api xmlns="http://ws.apache.org/ns/synapse"
+     name="$!apiName"
+     context="$!apiContext"
+     version="$!apiVersion"
+     transports="$!transport"
+     version-type="context"
+     #if($apiType == 'WS')
+     binds-to="WebSocketInboundEndpoint"
```
Comment (Nirothipan) [inline modules/distribution/resources/api_templates/velocity_template.xml:135]: how do we differentiate ws and wss , here

File: modules/distribution/resources/api_templates/default_ws_api_template.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<api xmlns="http://ws.apache.org/ns/synapse"  name="$!apiName" context="$!apiContext">
```
Comment (Nirothipan) [inline modules/distribution/resources/api_templates/default_ws_api_template.xml:1]: this needs to have binds-to ?

File: modules/distribution/resources/api_templates/velocity_template.xml (hunk 1 above)
Comment (senthuran16) [reply modules/distribution/resources/api_templates/velocity_template.xml:135]: Even if it's `WSS`, the type set will be `WS`, and it falls into the same case.

File: modules/distribution/resources/api_templates/websub_api_template.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+<api xmlns="http://ws.apache.org/ns/synapse" name="$!apiName" context="$!apiContext" version="$!apiVersion" version-type="context">
+    <resource methods="POST" url-mapping="/webhooks_events_receiver_resource">
```
Comment (Nirothipan) [inline modules/distribution/resources/api_templates/websub_api_template.xml:2]: let's have an _ at the start of this resource, @chanikag ?

File: modules/distribution/resources/api_templates/websub_api_template.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                    </sequence>
+                </target>
+            </clone>
+        </inSequence>
+    </resource>
+    <handlers>
+        <handler class="org.wso2.carbon.apimgt.gateway.handlers.streaming.webhook.WebhookApiHandler">
+            <property name="subscriptionResourcePath" value="/subscription"/>
+            <property name="topicQueryParamName" value="hub.secret"/>
+        </handler>
```
Comment (Nirothipan) [inline modules/distribution/resources/api_templates/websub_api_template.xml:78]: this need to have authentication handler as well, @chanikag  ?

File: modules/distribution/resources/api_templates/websub_api_template.xml [hunk 5]
```diff
@@ -0,0 +1,80 @@
+<api xmlns="http://ws.apache.org/ns/synapse" name="$!apiName" context="$!apiContext" version="$!apiVersion" version-type="context">
```
Comment (Nirothipan) [inline modules/distribution/resources/api_templates/websub_api_template.xml:1]: we need to have binds-t in api level as well as resource level

File: modules/distribution/resources/api_templates/websub_api_template.xml (hunk 4 above)
Comment (chanikag) [reply modules/distribution/resources/api_templates/websub_api_template.xml:78]: yes in addition to the rest of handlers we need to include this.

File: modules/distribution/resources/api_templates/default_ws_api_template.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<api xmlns="http://ws.apache.org/ns/synapse"  name="$!apiName" context="$!apiContext" binds-to="WebSocketInboundEndpoint">
```
Comment (Nirothipan) [inline modules/distribution/resources/api_templates/default_ws_api_template.xml:1]: let's add both binds-to this also ?

File: modules/distribution/resources/api_templates/velocity_template.xml (hunk 1 above)
Comment (senthuran16) [reply modules/distribution/resources/api_templates/velocity_template.xml]: Added `SecureWebSocketEP` to `binds-to` as discussed offline.
Addressed in https://github.com/wso2/product-apim/pull/9885/commits/19ebe0528979bdc4ced4a3abf78d4186f15a41bb

File: modules/distribution/resources/api_templates/default_ws_api_template.xml (hunk 6 above)
Comment (senthuran16) [reply modules/distribution/resources/api_templates/default_ws_api_template.xml]: Addressed in https://github.com/wso2/product-apim/pull/9885/commits/19ebe0528979bdc4ced4a3abf78d4186f15a41bb

File: modules/distribution/resources/api_templates/default_ws_api_template.xml (hunk 2 above)
Comment (senthuran16) [reply modules/distribution/resources/api_templates/default_ws_api_template.xml]: Addressed in https://github.com/wso2/product-apim/pull/9885/commits/19ebe0528979bdc4ced4a3abf78d4186f15a41bb

File: modules/distribution/resources/api_templates/websub_api_template.xml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                        <class name="org.wso2.carbon.apimgt.gateway.mediators.webhooks.DeliveryDataMediator"/>
+                        <drop/>
+                    </sequence>
+                </target>
+            </clone>
+        </inSequence>
+    </resource>
+    <handlers>
+        <handler class="org.wso2.carbon.apimgt.gateway.handlers.streaming.webhook.WebhookApiHandler">
+            <property name="subscriptionResourcePath" value="/subscription"/>
```
Comment (Nirothipan) [inline modules/distribution/resources/api_templates/websub_api_template.xml:76]: for the value of this one ( /subscription ), please use the same as the resource name in line 2

File: modules/distribution/resources/api_templates/websub_api_template.xml (hunk 4 above)
Comment (Nirothipan) [reply modules/distribution/resources/api_templates/websub_api_template.xml:78]: I think we might need cors as well, @chanikag ?

Comment (chanikag) [reply modules/distribution/resources/api_templates/websub_api_template.xml:78]: yeah all other handlers are applicable.

File: modules/distribution/resources/api_templates/websub_api_template.xml (hunk 7 above)
Comment (senthuran16) [reply modules/distribution/resources/api_templates/websub_api_template.xml]: Addressed in https://github.com/wso2/product-apim/pull/9885/commits/050bf78f67bc9c2951e815d07c2777fcd80bb627

File: modules/distribution/resources/api_templates/websub_api_template.xml (hunk 4 above)
Comment (senthuran16) [reply modules/distribution/resources/api_templates/websub_api_template.xml:118]: Addressed in https://github.com/wso2/product-apim/pull/9885/commits/050bf78f67bc9c2951e815d07c2777fcd80bb627

File: modules/distribution/resources/api_templates/websub_api_template.xml (hunk 5 above)
Comment (senthuran16) [reply modules/distribution/resources/api_templates/websub_api_template.xml]: Addressed in https://github.com/wso2/product-apim/pull/9885/commits/d54c96208f535aff0dacd4c3f9bcde96399d7f7d

**Review by Nirothipan — APPROVED** [review]

**Review by sajithshn — APPROVED** [review]

**Pull request merged**